src/serial.rs

```diff
@@ -101,7 +140,13 @@ macro_rules! hal {
         ),
     )+) => {
         $(
+            $(#[$meta])*
             impl<PINS> Serial<$USARTX, PINS> {
+
+            /// Configures the Serial interface and creates an interface struct.
+            /// `baud_rate` configures the baud rate of the interface. The other parameters are
+            /// handles to the corresponding registers, to make Rust's borrow checker happy.
+            /// (Ownership must be passed to this function)
```

Comment:
I would prefer "Takes ownership of the usart device" or something along those lines